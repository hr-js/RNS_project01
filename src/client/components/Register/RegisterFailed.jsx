// @flow
import * as React from "react";
import SingleButton from "@components/SingleButton";
import styles from "./style.css";

type RegisterFailedType = {
  onBackButton: (e: SyntheticEvent<HTMLButtonElement>) => {}
};

function RegisterFailed(props: RegisterFailedType): React.Node {
  const { onBackButton } = props;
  const { main, textOnly, contentWithButton } = styles;
  return (
    <div className={main}>
      <div className={`${contentWithButton} ${textOnly}`}>
        登録に失敗しました
      </div>
      <SingleButton
        text="入力画面に戻る"
        className="default"
        onButtonClick={onBackButton}
      />
    </div>
  );
}
export default RegisterFailed;
