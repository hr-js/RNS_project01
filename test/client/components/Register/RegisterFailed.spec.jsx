import React from "react";
import renderer from "react-test-renderer";
import { shallow, mount } from "enzyme";
import RegisterFailed from "@components/Register/RegisterFailed";

describe("RegisterFailed.jsxのテスト", () => {
  describe("スナップショット", () => {
    it("正しいレンダリング", () => {
      const tree = renderer.create(<RegisterFailed onBackButton={() => {}} />);
      expect(tree).toMatchSnapshot();
    });
  });

  describe("コンポーネントのテスト", () => {
    const onButton = () => {};
    const scan = shallow(<RegisterFailed onBackButton={onButton} />);

    it("cssクラス: mainをもつ", () => {
      expect(scan.hasClass("main")).toBeTruthy();
    });

    it("子要素のdivは、cssクラス: contentWithButtonとtextOnlyをもつ", () => {
      expect(
        scan
          .children()
          .find("div")
          .hasClass("contentWithButton")
      ).toBeTruthy();
      expect(
        scan
          .children()
          .find("div")
          .hasClass("textOnly")
      ).toBeTruthy();
    });

    it("「登録に失敗しました」と表示されている", () => {
      expect(
        scan
          .children()
          .find("div")
          .text()
      ).toBe("登録に失敗しました");
    });

    it("子要素のSingleButtonに、値を渡している", () => {
      expect(scan.find("SingleButton").prop("text")).toBe("入力画面に戻る");
      expect(scan.find("SingleButton").prop("className")).toBe("default");
      expect(scan.find("SingleButton").prop("onButtonClick")).toBe(onButton);
    });

    describe("SingleButtonをクリックした時", () => {
      it("onCancelButtonに渡した関数が呼び出される", () => {
        const mock = jest.fn();
        const fullDom = mount(<RegisterFailed onBackButton={mock} />);
        fullDom.find("SingleButton button").simulate("click");
        expect(mock).toBeCalled();
      });
    });
  });
});
