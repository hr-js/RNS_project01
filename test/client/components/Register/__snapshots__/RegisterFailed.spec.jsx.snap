// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterFailed.jsxのテスト スナップショット 正しいレンダリング 1`] = `
<div
  className="main"
>
  <div
    className="contentWithButton textOnly"
  >
    登録に失敗しました
  </div>
  <div
    className="singleButton"
  >
    <button
      className="default"
      onClick={[Function]}
    >
      入力画面に戻る
    </button>
  </div>
</div>
`;
